pandas/tests/io/json/test_ujson.py

```diff
@@ -720,6 +720,21 @@ def my_obj_handler(_):
             ujson.encode(obj_list, default_handler=str)
         )
 
+    def test_encode_object(self):
+        # Keys should be all non-callable non-underscore attributes, see GH-42768
+        class _TestObject:
+            def __init__(self, a, b, _c, d):
+                self.a = a
+                self.b = b
+                self._c = _c
+                self.d = d
+
+            def e(self):
+                return 5
+
+        test_object = _TestObject(a=1, b=2, _c=3, d=4)
+        assert ujson.decode(ujson.encode(test_object)) == {"a": 1, "b": 2, "d": 4}
```

Comment:
This is a great test - can we add an analogous test that hits `to_json` to test the user-facing effect?

Reply:
Sure, done.